src/charm.py

```diff
@@ -163,17 +194,17 @@ def _on_get_certificate_action(self, event: ActionEvent) -> None:
 
         event.set_results(
             {
-                f"certificate-{hostname}": tls_rel_data.get(f"certificate-{hostname}"),
-                f"ca-{hostname}": tls_rel_data.get(f"ca-{hostname}"),
-                f"chain-{hostname}": tls_rel_data.get(f"chain-{hostname}"),
+                f"certificate-{hostname}": tls_rel_data[f"certificate-{hostname}"],
+                f"ca-{hostname}": tls_rel_data.get[f"ca-{hostname}"],
+                f"chain-{hostname}": tls_rel_data.get[f"chain-{hostname}"],
             }
         )
 
     @block_if_invalid_config_or_missing_integration(defer=True)
     def _on_certificates_relation_created(self, _: RelationCreatedEvent) -> None:
         """Handle the TLS Certificate relation created event."""
         tls_information = TLSInformation.from_charm(self)
-        config = CharmConfig.from_charm(self)
+        config = CharmConfig.from_charm(self, get_client(self.app.name, self.model.name))
```

Comment:
Minor: I prefer to have a separate variable for method calls and not include them as arguments, as this makes debugging easier.

Reply:
Makes sense, I created separate variables in relevant places. It also makes it clearer that `self.app.name` is used as field_manager and `self.model.name` as k8s namespace.